Updated For Maven

agrosner opened the pull request:
1. Added gradle properties for each sub project for versioning. 
2. Moved files into gradle structure
3. Published version 1.0.0 of each to [maven-releases](https://github.com/Raizlabs/maven-releases)
4. Artifacts include: Adapters, BaseUtils, ImageCaching, and JsonUtils

Commit:
moved files into gradle structure, added gradle.properties. next will be to add publishing mechanism

Commit:
added artifacts files, seeing if this works

Commit:
added wrapper

Commit:
trying this

Commit:
fix

Commit:
trying this

Commit:
Merge remote-tracking branch 'origin/splitLibraries' into maven

dylanrjames commented on android-artifacts.gradle:
```diff
@@ -0,0 +1,22 @@
+
+    artifacts {
+        archives androidSourcesJar
+        //archives androidJavadocsJar
```
Any reason to leave this commented stuff here? And any reason why we wouldn't build Javadocs? Don't we need to do this for Javadoc to be visible from consuming code?

agrosner replied:
Dylan, I am not sure what the correct gradle variable was and never got
around to finding it out. But by running that command, the gradle build
will fail.

On Fri, Jan 23, 2015 at 10:33 AM, dylanrjames notifications@github.com
wrote:

> In android-artifacts.gradle
> https://github.com/Raizlabs/RZAndroidBaseUtils/pull/6#discussion_r23456510
> :
> 
> > +
> > -   task androidJavadocsJar(type: Jar, dependsOn: androidJavadocs) {
> > -       classifier = 'javadoc'
> > -       //basename = artifact_id
> > -       from androidJavadocs.destinationDir
> > -   }*/
> >   +
> > -    task androidSourcesJar(type: Jar) {
> > -        classifier = 'sources'
> > -        //basename = artifact_id
> > -        from android.sourceSets.main.java.sourceFiles
> > -    }
> >   +
> > -    artifacts {
> > -        archives androidSourcesJar
> > -        //archives androidJavadocsJar
> 
> Any reason to leave this commented stuff here? And any reason why we
> wouldn't build Javadocs? Don't we need to do this for Javadoc to be visible
> from consuming code?
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/Raizlabs/RZAndroidBaseUtils/pull/6/files#r23456510.

## 

## 

_Andrew Grosner, Android Developer_

Mobile Strategy, Design & Development

www.raizlabs.com | Check my calendar
https://www.google.com/calendar/embed?src=andrew.grosner@raizlabs.com&ctz=America/New_York&mode=WEEK
 | @agrosner http://www.twitter.com/agrosner | 716-472-5075
[image: Raizlabs]

agrosner replied:
I left it in there in case someday I will find what it was and this file
was a simple copy and paste from the master file that I have.

On Fri, Jan 23, 2015 at 10:35 AM, Andrew Grosner <
andrew.grosner@raizlabs.com> wrote:

> Dylan, I am not sure what the correct gradle variable was and never got
> around to finding it out. But by running that command, the gradle build
> will fail.
> 
> On Fri, Jan 23, 2015 at 10:33 AM, dylanrjames notifications@github.com
> wrote:
> 
> > In android-artifacts.gradle
> > https://github.com/Raizlabs/RZAndroidBaseUtils/pull/6#discussion_r23456510
> > :
> > 
> > > +
> > > -   task androidJavadocsJar(type: Jar, dependsOn: androidJavadocs) {
> > > -       classifier = 'javadoc'
> > > -       //basename = artifact_id
> > > -       from androidJavadocs.destinationDir
> > > -   }*/
> > >   +
> > > -    task androidSourcesJar(type: Jar) {
> > > -        classifier = 'sources'
> > > -        //basename = artifact_id
> > > -        from android.sourceSets.main.java.sourceFiles
> > > -    }
> > >   +
> > > -    artifacts {
> > > -        archives androidSourcesJar
> > > -        //archives androidJavadocsJar
> > 
> > Any reason to leave this commented stuff here? And any reason why we
> > wouldn't build Javadocs? Don't we need to do this for Javadoc to be visible
> > from consuming code?
> > 
> > —
> > Reply to this email directly or view it on GitHub
> > https://github.com/Raizlabs/RZAndroidBaseUtils/pull/6/files#r23456510.
> 
> ## 
> 
> ## 
> 
> _Andrew Grosner, Android Developer_
> 
> Mobile Strategy, Design & Development
> 
> www.raizlabs.com | Check my calendar
> https://www.google.com/calendar/embed?src=andrew.grosner@raizlabs.com&ctz=America/New_York&mode=WEEK
>  | @agrosner http://www.twitter.com/agrosner | 716-472-5075
> [image: Raizlabs]

## 

## 

_Andrew Grosner, Android Developer_

Mobile Strategy, Design & Development

www.raizlabs.com | Check my calendar
https://www.google.com/calendar/embed?src=andrew.grosner@raizlabs.com&ctz=America/New_York&mode=WEEK
 | @agrosner http://www.twitter.com/agrosner | 716-472-5075
[image: Raizlabs]

dylanrjames replied:
Does this mean that the JavaDoc doesn't appear when you pull it in as a Gradle dependency like this?

agrosner replied:
It means they're not generated as a javadoc jar. The sources still appear
and thus we can link the sources which also contain the comments and
javadoc information.

On Fri, Jan 23, 2015 at 10:38 AM, dylanrjames notifications@github.com
wrote:

> In android-artifacts.gradle
> https://github.com/Raizlabs/RZAndroidBaseUtils/pull/6#discussion_r23456836
> :
> 
> > +
> > -   task androidJavadocsJar(type: Jar, dependsOn: androidJavadocs) {
> > -       classifier = 'javadoc'
> > -       //basename = artifact_id
> > -       from androidJavadocs.destinationDir
> > -   }*/
> >   +
> > -    task androidSourcesJar(type: Jar) {
> > -        classifier = 'sources'
> > -        //basename = artifact_id
> > -        from android.sourceSets.main.java.sourceFiles
> > -    }
> >   +
> > -    artifacts {
> > -        archives androidSourcesJar
> > -        //archives androidJavadocsJar
> 
> Does this mean that the JavaDoc doesn't appear when you pull it in as a
> Gradle dependency like this?
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/Raizlabs/RZAndroidBaseUtils/pull/6/files#r23456836.

## 

## 

_Andrew Grosner, Android Developer_

Mobile Strategy, Design & Development

www.raizlabs.com | Check my calendar
https://www.google.com/calendar/embed?src=andrew.grosner@raizlabs.com&ctz=America/New_York&mode=WEEK
 | @agrosner http://www.twitter.com/agrosner | 716-472-5075
[image: Raizlabs]

dylanrjames replied:
Still not sure I'm understanding this - let's talk about this one in person.

agrosner replied:
You're looking at the wrong repo BTW it's CoreUtils
On Mar 29, 2015 8:51 PM, "dylanrjames" notifications@github.com wrote:

> In android-artifacts.gradle
> https://github.com/Raizlabs/RZAndroidBaseUtils/pull/6#discussion_r27361767
> :
> 
> > +
> > -   task androidJavadocsJar(type: Jar, dependsOn: androidJavadocs) {
> > -       classifier = 'javadoc'
> > -       //basename = artifact_id
> > -       from androidJavadocs.destinationDir
> > -   }*/
> >   +
> > -    task androidSourcesJar(type: Jar) {
> > -        classifier = 'sources'
> > -        //basename = artifact_id
> > -        from android.sourceSets.main.java.sourceFiles
> > -    }
> >   +
> > -    artifacts {
> > -        archives androidSourcesJar
> > -        //archives androidJavadocsJar
> 
> Still not sure I'm understanding this - let's talk about this one in
> person.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/Raizlabs/RZAndroidBaseUtils/pull/6/files#r27361767.

dylanrjames commented:
A couple general high-level things:

1) Do we want to actually keep these in the same repo? Is there an advantage to doing this? I was expecting breaking these out into separate Git repos because they're independent chunks of code. Would also help us with avoiding conflicts etc.

2) (Possibly related to 1) It seems like the projects are referencing dependencies locally (namely Adapters -> BaseUtils). This keeps us relatively constrained on the folder structure, which works if we leave them in one repo, but reduces flexibility. Wouldn't we rather do these as Gradle/Maven/remote dependencies so this is not the case?

3) Is there any reason to be moving to the Gradle project structure? As is, I don't see any advantage and we're possibly breaking old projects/setups in Eclipse. I'd be fine doing this on new projects but I'm not sure I want to migrate over old ones unless there's a reason to.

agrosner commented:
1. No I had kept it there because I thought the plan was to keep them in
   the same repo for now. Yes, we can totally break them apart into new repos.
   If you were to add those today we could get them up and running.
2. No real reason, when we split these up, we should make it remote anyways.
3. If we split up BaseUtils, then we can leave this repo as such, remove
   this pull request and just use the code from this branch to create the
   other repos. The reason for the structure change is have it match the
   gradle Android standard structure. It also simplifies the build.gradle
   file. Since google is not really supporting Eclipse anymore, it does not
   make much sense for the new repos to support the old structure anyways. And
   if I remember correctly, you can "trick" Eclipse into using the new
   structure for repos.

On Fri, Jan 23, 2015 at 10:58 AM, dylanrjames notifications@github.com
wrote:

> A couple general high-level things:
> 
> 1) Do we want to actually keep these in the same repo? Is there an
> advantage to doing this? I was expecting breaking these out into separate
> Git repos because they're independent chunks of code. Would also help us
> with avoiding conflicts etc.
> 
> 2) (Possibly related to 1) It seems like the projects are referencing
> dependencies locally (namely Adapters -> BaseUtils). This keeps us
> relatively constrained on the folder structure, which works if we leave
> them in one repo, but reduces flexibility. Wouldn't we rather do these as
> Gradle/Maven/remote dependencies so this is not the case?
> 
> 3) Is there any reason to be moving to the Gradle project structure? As
> is, I don't see any advantage and we're possibly breaking old
> projects/setups in Eclipse. I'd be fine doing this on new projects but I'm
> not sure I want to migrate over old ones unless there's a reason to.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/Raizlabs/RZAndroidBaseUtils/pull/6#issuecomment-71214617
> .

## 

## 

_Andrew Grosner, Android Developer_

Mobile Strategy, Design & Development

www.raizlabs.com | Check my calendar
https://www.google.com/calendar/embed?src=andrew.grosner@raizlabs.com&ctz=America/New_York&mode=WEEK
 | @agrosner http://www.twitter.com/agrosner | 716-472-5075
[image: Raizlabs]

dylanrjames commented:
I guess a lot of this just comes down to whether we want to split these up into different repos and artifacts. I'm leaning towards yes. Seems like you agree?

You can "trick" eclipse to grab a different code source folder, but I don't think you can do resources etc. If we migrate the project structure, I'd just throw in the towel for supporting Eclipse on them. Which makes me feel like we should do a reversioning and make all of our major changes at once.

agrosner commented:
Yeah agreed. Lets split them up! I may work on my dependency plugin to
support the link sources plugin automatically. I want to have to avoid
something like this in our projects:
  apt 'com.raizlabs.android:Broker-Compiler:1.1.0'
  aarLinkSources 'com.raizlabs.android:Broker-Compiler:1.1.0:sources@jar'
  compile 'com.raizlabs.android:Broker-Core:1.1.0'
  aarLinkSources 'com.raizlabs.android:Broker-Core:1.1.0:sources@jar'
  compile 'com.raizlabs.android:Broker: 1.1.0'
  aarLinkSources 'com.raizlabs.android:Broker:1.1.0:sources@jar'
  compile 'com.raizlabs.android:Broker-Volley:1.1.0'
  aarLinkSources 'com.raizlabs.android:Broker-Volley:1.1.0:sources@jar'
  compile 'com.raizlabs.android:Broker-WebServiceManager:1.1.0'
  aarLinkSources
'com.raizlabs.android:Broker-WebServiceManager:1.1.0:sources@jar'

On Fri, Jan 23, 2015 at 11:45 AM, dylanrjames notifications@github.com
wrote:

> I guess a lot of this just comes down to whether we want to split these up
> into different repos and artifacts. I'm leaning towards yes. Seems like you
> agree?
> 
> You can "trick" eclipse to grab a different code source folder, but I
> don't think you can do resources etc. If we migrate the project structure,
> I'd just throw in the towel for supporting Eclipse on them. Which makes me
> feel like we should do a reversioning and make all of our major changes at
> once.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/Raizlabs/RZAndroidBaseUtils/pull/6#issuecomment-71223119
> .

## 

## 

_Andrew Grosner, Android Developer_

Mobile Strategy, Design & Development

www.raizlabs.com | Check my calendar
https://www.google.com/calendar/embed?src=andrew.grosner@raizlabs.com&ctz=America/New_York&mode=WEEK
 | @agrosner http://www.twitter.com/agrosner | 716-472-5075
[image: Raizlabs]

dylanrjames commented on gradle.properties:
```diff
@@ -0,0 +1,24 @@
+POM_LICENCE_URL=http://opensource.org/licenses/MIT
+POM_LICENCE_DIST=repo
+POM_DEVELOPER_ID=agrosner
+POM_DEVELOPER_NAME=Andrew Grosner
```
These should be shared or Raizlabs IDs instead of personal ones.

dylanrjames commented:
Since we're making significant changes to this, we should reversion this as a 2.0. This is breaking backwards compatibility for Eclipse etc.

agrosner commented:
@dylanrjames you're looking at the wrong repo. Check out CoreUtils
